# Copyright 2026 Intrinsic Innovation LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Manually removes `qmake` and `mkspecs/linux-g++`, which are not available in our sysroot,
# but referenced in `Qt5Config.cmake`.
--- usr/lib/x86_64-linux-gnu/cmake/Qt5Core/Qt5CoreConfigExtras.cmake
+++ usr/lib/x86_64-linux-gnu/cmake/Qt5Core/Qt5CoreConfigExtras.cmake
@@ -2,17 +2,6 @@
     set(QT_DEFAULT_MAJOR_VERSION 5)
 endif()

-if (NOT TARGET Qt5::qmake)
-    add_executable(Qt5::qmake IMPORTED)
-
-    set(imported_location "${_qt5Core_install_prefix}/lib/x86_64-linux-gnu/qt5/bin/qmake")
-    _qt5_Core_check_file_exists(${imported_location})
-
-    set_target_properties(Qt5::qmake PROPERTIES
-        IMPORTED_LOCATION ${imported_location}
-    )
-endif()
-
 if (NOT TARGET Qt5::moc)
     add_executable(Qt5::moc IMPORTED)

@@ -39,7 +28,6 @@

 set(CMAKE_AUTOMOC_MACRO_NAMES Q_OBJECT Q_GADGET Q_NAMESPACE Q_NAMESPACE_EXPORT)

-set(Qt5Core_QMAKE_EXECUTABLE Qt5::qmake)
 set(Qt5Core_MOC_EXECUTABLE Qt5::moc)
 set(Qt5Core_RCC_EXECUTABLE Qt5::rcc)

@@ -49,16 +37,6 @@
   COMPATIBLE_INTERFACE_STRING QT_MAJOR_VERSION QT_COORD_TYPE
 )

-include("${CMAKE_CURRENT_LIST_DIR}/Qt5CoreConfigExtrasMkspecDir.cmake")
-
-foreach(_dir ${_qt5_corelib_extra_includes})
-    _qt5_Core_check_file_exists(${_dir})
-endforeach()
-
-list(APPEND Qt5Core_INCLUDE_DIRS ${_qt5_corelib_extra_includes})
-set_property(TARGET Qt5::Core APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${_qt5_corelib_extra_includes})
-set(_qt5_corelib_extra_includes)
-
 # Targets using Qt need to use the POSITION_INDEPENDENT_CODE property. The
 # Qt5_POSITION_INDEPENDENT_CODE variable is used in the # qt5_use_module
 # macro to add it.
@@ -97,7 +75,7 @@
 _qt5_Core_check_file_exists(${_Qt5CTestMacros})

 # Create versionless tool targets.
-foreach(__qt_tool qmake moc rcc)
+foreach(__qt_tool moc rcc)
     if(NOT "${QT_NO_CREATE_VERSIONLESS_TARGETS}" AND NOT TARGET Qt::${__qt_tool}
        AND TARGET Qt5::${__qt_tool})
         add_executable(Qt::${__qt_tool} IMPORTED)
